dags/utils/backoff.py

```diff
@@ -0,0 +1,19 @@
+import time
+import logging
+import random
+
+logger = logging.getLogger()
+
+def retry_with_backoff(fn, retries = 10, backoff_in_seconds = 1):
+    x = 0
+    while True:
+        try:
+            return fn()
+        except:
+            if x == retries:
+                raise
+            sleep = (backoff_in_seconds * 2 ** x +
+                    random.uniform(0, 1))
+            logger.critical("Attempt {}, retrying in {} seconds".format(x, round(sleep)))
+            time.sleep(sleep)
+        x += 1
```

Comment:
This could be better written with a recursive function to avoid a while loop
```python
    try:
      return fn()
    except:
      if retries <= 0:
        raise
      sleep ...
      retry_with_backoff(fn, retries - 1, backoff_in_seconds)
```

Reply:
The problem with this is the backoff is calculated by counting up to the retries number. With this implementation the algorithm would start with the maximum wait time which wouldn't make sense.